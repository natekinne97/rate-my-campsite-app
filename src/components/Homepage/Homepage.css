.landing-page{
    background: url('https://i.imgur.com/K8By51k.jpg');
    width: 100%;
    text-align: center;
    width: 100%;
	height: 90vh;
	background-size: cover;
    background-position: 50% 50%;
    color: white;
    margin-top: 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
}