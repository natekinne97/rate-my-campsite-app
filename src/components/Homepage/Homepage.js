import React from 'react'
import Featured from '../Featured/Featured';
import Campsite from '../Campsites/Campsite';
import './Homepage.css';


class Homepage extends React.Component{

    renderLandingPage(){
        return(<div className="landing-page">
            <header>
                <h2>Show the world your great campsite discovories</h2>
                <p>"A social media for campsites."</p>
                <p>Post your favorite campsites and let the world review them.</p>
                <p>Test user: c.bloggs</p>
                <p>Test password: charzard</p>
                <p>Login to post new campsites</p>
            </header>
        </div>);
    }

    render(){
        return(
            <React.Fragment>

                {this.renderLandingPage()}

                <Featured/>
                
                    <Campsite />
               
               
            </React.Fragment>
        );
    }
}

export default Homepage;