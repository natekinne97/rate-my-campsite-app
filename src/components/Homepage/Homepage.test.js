import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Homepage from './Homepage';

describe.only('Review form component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Homepage />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Homepage />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the UI as expected with no unreads', () => {
        const tree = renderer
            .create(<Homepage />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});