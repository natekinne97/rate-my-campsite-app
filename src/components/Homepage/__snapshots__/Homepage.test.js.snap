// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review form component renders the UI as expected 1`] = `
Array [
  <div
    className="landing-page"
  >
    <header>
      <h2>
        Show the world your great campsite discovories
      </h2>
      <p>
        "A social media for campsites."
      </p>
      <p>
        Post your favorite campsites and let the world review them.
      </p>
    </header>
  </div>,
  <div
    className="featured-shell"
  >
    <p
      className="red"
    >
      Error occured. Please try again later
    </p>
    <p
      className="red"
    >
      Error occured. Please try again later
    </p>
  </div>,
  <div
    className="all-posts"
  />,
]
`;

exports[`Review form component renders the UI as expected with no unreads 1`] = `
Array [
  <div
    className="landing-page"
  >
    <header>
      <h2>
        Show the world your great campsite discovories
      </h2>
      <p>
        "A social media for campsites."
      </p>
      <p>
        Post your favorite campsites and let the world review them.
      </p>
    </header>
  </div>,
  <div
    className="featured-shell"
  >
    <p
      className="red"
    >
      Error occured. Please try again later
    </p>
    <p
      className="red"
    >
      Error occured. Please try again later
    </p>
  </div>,
  <div
    className="all-posts"
  />,
]
`;
